eventer.go

```diff
@@ -1,36 +1,116 @@
 package gobot
 
+type eventChannel chan *Event
+
 type eventer struct {
-	events map[string]*Event
+	// map of valid Event names
+	eventnames map[string]string
+
+	// new events get put in to the event channel
+	in eventChannel
+
+	// slice of out channels used by subscribers
+	outs []eventChannel
 }
 
-// Eventer is the interface which describes behaviour for a Driver or Adaptor
-// which uses events.
+// Eventer is the interface which describes how a Driver or Adaptor
+// handles events.
 type Eventer interface {
```

Comment:
*Caveat for all this feedback: I need to get all my Gobot knowledge back into a higher-lvl mental cache than it's in right now, so please forgive me if some of this comes across as a bit out-of-date or ignorant.*

If I want to delete an event, it would seem that right now I would just call `AddEvent` in such a way as to force a key collision and reset the value in the map. I suggest having `DeleteEvent` for removing a key and to make `AddEvent` throw an error if you try to add one with the same name. Does that make sense for what you're trying to achieve with this interface?